util.sh

```diff
@@ -11,6 +11,48 @@ fi
 
 SOURCE=${BASH_SOURCE:-}
 
+# ============================================================================
+# GNU
+# ============================================================================
+
+# use gnu version of command if exists
+# usage:
+# gnu <command> <args>
+# example:
+# gnu grep foo
+function gnu {
+    command=$1
+    gnucommand=g$command
+    path=$(which $command 2> /dev/null || true)
+    gnupath=$(which $gnucommand 2> /dev/null || true)
+
+    shift
+
+    bin=${gnupath:-$path}
+
+    if [ -z "$bin" ]; then
+        echo -e ${RED}${command}${END_COLOR} or ${RED}${gnucommand}${END_COLOR} not found
+        exit 1
+    fi
+
+    if [ $(uname -s) = "Darwin" ] && [ -z "$gnupath" ]; then
+        echo -e Mac detected but ${RED}$gnucommand${END_COLOR} not found > /dev/stderr
+        echo -e Mac version of ${BLUE}${command}${END_COLOR} is usually very outdated and might not work as expected > /dev/stderr
+        echo -e Consider installing GNU version of ${BLUE}${command}${END_COLOR} with brew > /dev/stderr
+        echo -e Usually: > /dev/stderr
+        echo -e "\tbrew install ${command}" > /dev/stderr
+        echo > /dev/stderr
```

Comment:
this will show up as:

<img width="650" alt="image" src="https://user-images.githubusercontent.com/932940/184685375-d2885b3b-bb53-474f-ace3-79a8d6d0ce54.png">

(for the example changed to `xgrep` since I already have gnu versions installed

Reply:
Btw this issue is also appearing with the brew version of grep

Reply:
I think it needed a few more tools like `sed` from GNU. can you try this branch and let me know if this works as expected now?